db/controllers/articleCommentsController.js

```diff
@@ -21,5 +21,35 @@ exports.getArticleComments = (req, res, next) => {
 }
 
 
+exports.postComment = (req, res, next) => {
+    const newComment = req.body.body; 
+    const usernameReq = req.body.username 
+    const reqBodyKeys = Object.keys(req.body);
+    const usernameProp = reqBodyKeys[0];
+    const articleId = Object.values(req.params); 
+    const articleIdNum = Number(articleId);
+
+  
+    const promise1 = checkExists('article_id', 'articles', Number(articleId[0]));
+    const promise2 = checkExists('username', 'users', usernameReq);
+    const promise3 = checkValid(usernameProp, 'username');
+    
+   
+    Promise.all([promise1, promise2, promise3])
+    .then((fulfilledProms) => {
+        return fulfilledProms;
+    })
+    .then((fulfilledProms) => {
+        return insertComment(newComment, articleIdNum, usernameReq, fulfilledProms[0], fulfilledProms[1], fulfilledProms[2])
+    })
+    .then((comment) => {
+    res.status(201).send({ comment: comment });   
+    })    
+    .catch((err) => {
+            next(err)
+     });   
```

Comment:
This code is really over engineered, you don't need to make checks for each property. You should let PSQL handle these errors rather than creating custom errors.